fdbserver/include/fdbserver/LogSystem.h

```diff
@@ -834,13 +837,17 @@ struct LogPushData : NonCopyable {
 	// true on a successful write, and false if the location has already been
 	// written.
 	bool writeTransactionInfo(int location, uint32_t subseq);
+
+	Tag chooseRouterTag() {
+		return savedRandomRouterTag.present() ? savedRandomRouterTag.get() : logSystem->getRandomRouterTag();
```

Comment:
I am noticing that this is resulting in an assertion failure in "TagPartitionedLogSystem::getRandomRouterTag()" -> "deterministicRandom()->randomInt()" when "logRouterTags = 0".

Reply:
this appears to be because a pseudo tag is [set](https://github.com/apple/foundationdb/blob/5ee0db13e6151fb09ffef5845a3c3f88e06059ed/fdbserver/TagPartitionedLogSystem.actor.cpp#L1844).